htdocs/wp-content/themes/owc-formulieren/assets/js/editor/personal-heading/edit.js

```diff
@@ -0,0 +1,336 @@
+import { __ } from '@wordpress/i18n';
+import {
+	useBlockProps,
+	MediaUpload,
+	MediaUploadCheck,
+	RichText,
+	InspectorControls,
+} from '@wordpress/block-editor';
+import {
+	Card,
+	CardHeader,
+	CardBody,
+	CardDivider,
+	Button,
+	Panel,
+	PanelBody,
+	PanelRow,
+} from '@wordpress/components';
+
+const ALLOWED_MEDIA_TYPES = ['image'];
+
+const Edit = ( props ) => {
+	const blockProps = useBlockProps( {
+		className: 'openpdd-personal-heading',
+	} );
+
+	const { attributes, setAttributes } = props;
+	const {
+		textContent,
+		bgImageMorningId,
+		bgImageMorningUrl,
+		bgImageAfternoonId,
+		bgImageAfternoonUrl,
+		bgImageEveningId,
+		bgImageEveningUrl,
+	} = attributes;
+
+	const fallbackInstructions =
+		<p>{__( 'U heeft geen rechten om foto\'s aan te passen. Neem contact op met de beheerder.', 'owc-formulieren' )}</p>;
+
+	const onUpdateMorningImage = ( image ) => {
+		let imageUrl = image.sizes.large.url || image.url;
+
+		setAttributes( {
+			bgImageMorningId: image.id,
+			bgImageMorningUrl: imageUrl,
+		} );
+	};
+
+	const onRemoveMorningImage = () => {
+		setAttributes( {
+			bgImageMorningId: undefined,
+			bgImageMorningUrl: undefined,
+		} );
+	};
+
+	const onUpdateAfternoonImage = ( image ) => {
+		let imageUrl = image.sizes.large.url || image.url;
+
+		setAttributes( {
+			bgImageAfternoonId: image.id,
+			bgImageAfternoonUrl: imageUrl,
+		} );
+	};
+
+	const onRemoveAfternoonImage = () => {
+		setAttributes( {
+			bgImageAfternoonId: undefined,
+			bgImageAfternoonUrl: undefined,
+		} );
+	};
+
+	const onUpdateEveningImage = ( image ) => {
+		let imageUrl = image.sizes.large.url || image.url;
+
+		setAttributes( {
+			bgImageEveningId: image.id,
+			bgImageEveningUrl: imageUrl,
+		} );
+	};
+
+	const onRemoveEveningImage = () => {
+		setAttributes( {
+			bgImageEveningId: undefined,
+			bgImageEveningUrl: undefined,
+		} );
+	};
+
+	const onContentChange = ( content ) => {
+		setAttributes( {
+			textContent: content,
+		} );
+	};
+
+	return (
+		<div {...blockProps}>
+			<InspectorControls>
+				<Panel header={__( 'Persoonlijke heading', 'owc-formulieren' )}>
+					<PanelBody
```

Comment:
Veel beter zo!

Er kunnen nog wat componenten uit worden gehaald. Per foto een `<PanelBody/>`, en dan zou dit voldoende moeten zijn: 

```
<Panel header={__('Persoonlijke heading', 'owc-formulieren')}>
	<PanelBody
		title={__('Ochtend foto', 'owc-formulieren')}
		initialOpen={true}
	>
		<MediaUploadCheck fallback={fallbackInstructions}>
			<MediaUpload
				title={__('Foto ochtend', 'owc-formulieren')}
				onSelect={onUpdateMorningImage}
				allowedTypes={ALLOWED_MEDIA_TYPES}
				value={bgImageMorningId}
				render={({ open }) => (
					<Button
						className={
							!bgImageMorningId
								? 'editor-post-featured-image__toggle'
								: 'editor-post-featured-image__preview'
						}
						onClick={open}
					>
						{!bgImageMorningId &&
							__('Selecteer een ochtend foto', 'owc-formulieren')}
						{!!bgImageMorningId &&
							bgImageMorningUrl && (
								<img
									src={bgImageMorningUrl}
									alt={__('Foto ochtend', 'owc-formulieren')}
								/>
							)}
					</Button>
				)}
			/>
		</MediaUploadCheck>
		{!!bgImageMorningId && bgImageMorningUrl && (
			<MediaUploadCheck>
				<MediaUpload
					title={__('Ochtend foto', 'owc-formulieren')}
					onSelect={onUpdateMorningImage}
					allowedTypes={ALLOWED_MEDIA_TYPES}
					value={bgImageMorningId}
					render={({ open }) => (
						<Flex justify="flex-start" gap={3}>
							<Button onClick={open} isDefault>
								{__('Vervang de foto', 'owc-formulieren')}
							</Button>
							<Button
								onClick={onRemoveMorningImage}
								isLink
								isDestructive
							>
								{__('Verwijder foto', 'owc-formulieren')}
							</Button>
						</Flex>
					)}
				/>
			</MediaUploadCheck>
		)}
	</PanelBody>
 // En dan verder een <PanelBody /> voor de middagfoto en nogmaals een <PanelBody /> voor de avondfoto. 
```

Reply:
Je kan het nog meer abstraheren door een apart component te maken van de ImageUplaod. Dan heb je nog minder duplicated code. 

```
const ImageUpload = ({ timeOfDay, imageUrl, onUpload, onRemove }) => {
  return (
   <PanelBody title={`${timeOfDay} foto`} ... >
      ...
  )
}
```
En dan 
```
<ImageUpload
  timeOfDay="Ochtend"
  imageUrl={bgImageMorningUrl}
  onUpload={(image) => onUpdateImage('Morning', image)}
  onRemove={() => onRemoveImage('Morning')}
/>
<ImageUpload
  timeOfDay="Middag"
  imageUrl={bgImageAfternoonUrl}
  onUpload={(image) => onUpdateImage('Afternoon', image)}
  onRemove={() => onRemoveImage('Afternoon')}
/>
<ImageUpload
  timeOfDay="Avond"
  imageUrl={bgImageEveningUrl}
  onUpload={(image) => onUpdateImage('Evening', image)}
  onRemove={() => onRemoveImage('Evening')}
/>
```

Reply:
Kijk maar of bovenstaande lukt. Zo niet, dan is het geen ramp.